### PR TITLE
feat(config): add config file for Shenzhen Neo PIR Motion, Temperature & Light Sensor

### PR DESCRIPTION
**Pull request by dannybloe**
Support for Shenzhen Neo Electronics Co., Ltd. new PIR Motion, Temperature & Light Sensor.
This one is different in many ways from the previous NAS_PD02Z. Hence the seperate json.

### CODE REVIEW

File: packages/config/config/devices/0x0258/nas-pd02z-2.json [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+		}
+	],
+	"firmwareVersion": {
+		"min": "0.0",
+		"max": "255.255"
+	},
+	"associations": {
+		"1": {
+			"label": "Lifeline",
+			"description": "Group 1 is lifeline group; all nodes which associated in this group will receive the messages sent by device through lifeline.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0258/nas-pd02z-2.json]: We don't actually use the description. Should probably remove it from the import script.
```suggestion
```

File: packages/config/config/devices/0x0258/nas-pd02z-2.json [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+			"maxNodes": 1,
+			"isLifeline": true
+		},
+		"2": {
+			"label": "Control",
+			"description": "Group 2 is controlling group, all nodes associated in this group will be controlled through BASIC_SET command by the device when device detects a movement event.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0258/nas-pd02z-2.json]: dito, also lets rename the group to be clearer
```suggestion
			"label": "Motion Control",
```

File: packages/config/config/devices/0x0258/nas-pd02z-2.json [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+			"$import": "templates/shenzhen_neo_template.json#blink_on_motion"
+		},
+		{
+			"#": "8",
+			"label": "Binary Sensor Report Enable",
+			"$import": "~/templates/master_template.json#base_enable_disable"
+		},
+		{
+			"#": "9",
+			"label": "Motion Sensor: Basic Set Level",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0258/nas-pd02z-2.json]:
```suggestion
			"label": "Motion Control: Basic Set Level",
```
to make a better correlation to the association group

File: packages/config/config/devices/0x0258/nas-pd02z-2.json [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+			"description": "Value for BASIC_SET for nodes associated in Group 2. '100'- BASIC_SET = 0xFF (ON). '0'- BASIC_SET = 0x00 (OFF).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0258/nas-pd02z-2.json]: This description should be replaced with options for 0 and 100/0xff

File: packages/config/config/devices/0x0258/nas-pd02z-2.json [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+			"valueSize": 1,
+			"minValue": 1,
+			"maxValue": 100,
+			"defaultValue": 100
+		},
+		{
+			"#": "10",
+			"label": "Light Sensor: Measuring Interval",
+			"description": "Time interval for light sensor measurement of ambient light intensity. Higher values result in longer battery life.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0258/nas-pd02z-2.json]: description seems unnecessary

File: packages/config/config/devices/0x0258/nas-pd02z-2.json [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+			"unit": "seconds",
+			"valueSize": 2,
+			"minValue": 30,
+			"maxValue": 3600,
+			"defaultValue": 180
+		},
+		{
+			"#": "11",
+			"label": "Light Sensor: Intensity Differential Report",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0258/nas-pd02z-2.json]: does this have a unit?
I think we typically call this `Brightness Change Report Threshold` or similar.

File: packages/config/config/devices/0x0258/nas-pd02z-2.json [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+			"valueSize": 1,
+			"minValue": 1,
+			"maxValue": 127,
+			"defaultValue": 50
+		},
+		{
+			"#": "12",
+			"label": "Light Sensor: Intensity Threshold",
+			"description": "If the light intensity current measured is less than the settings, device will considered to be dark at current time. If configuration 13 is enabled and a motion event is detected, the device will send a BASIC_SET to the nodes that associated in group 2.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0258/nas-pd02z-2.json]: AFAIK other devices call this Night/Day Threshold. We should do the same here and update the corresponding parameters accordingly.

File: packages/config/config/devices/0x0258/nas-pd02z-2.json [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+			"#": "13",
+			"label": "Light Sensor: Intensity Associated Enable",
+			"description": "Send BASIC_SET commands to nodes in group 2 only if the environment is 'dark' (as defined by 12: 'Light Intensity Threshold').",
+			"$import": "~/templates/master_template.json#base_enable_disable"
+		},
+		{
+			"#": "14",
+			"label": "Motion Sensor: Report Once Enable",
+			"description": "Suppress sending more Motion Event Report messages after the first Motion Event Report message was sent until a Motion Event Cleared Report message has been sent.",
+			"$import": "~/templates/master_template.json#base_enable_disable"
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0258/nas-pd02z-2.json]: `$import` should be the first thing after the param number

File: packages/config/config/devices/0x0258/nas-pd02z-2.json [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+		},
+		{
+			"#": "15",
+			"label": "Temperature Sensor: Differential Report",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0258/nas-pd02z-2.json]:
```suggestion
			"label": "Temperature Change Report Threshold",
```
(like above)

File: packages/config/config/devices/0x0258/nas-pd02z-2.json [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+			"maxValue": 3600,
+			"defaultValue": 30
+		},
+		{
+			"#": "7",
+			"$import": "templates/shenzhen_neo_template.json#blink_on_motion"
+		},
+		{
+			"#": "8",
+			"label": "Binary Sensor Report Enable",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0258/nas-pd02z-2.json]: We try not to call parameters "... Enable" or "... Disable".

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Couple of things. Also in general the descriptions are quite long, please try to shorten them as much as possible and remove redundant information that is available through the other properties (unit, range, ...).

File: packages/config/config/devices/0x0258/nas-pd02z-2.json (hunk 1 above)
Comment (dannybloe) [reply packages/config/config/devices/0x0258/nas-pd02z-2.json]: Done

File: packages/config/config/devices/0x0258/nas-pd02z-2.json (hunk 6 above)
Comment (dannybloe) [reply packages/config/config/devices/0x0258/nas-pd02z-2.json]: It seems to be called 'Illumination Differential Report' everywhere.

**Comment by dannybloe** [comment]
I think I addressed all issues.

**Review by AlCalzone — APPROVED** [review]

**Comment by AlCalzone** [comment]
I **knew** I'd seen those parameters before. That device is the same as INT-SMMD-N1. I'll align those at some point, but this is good for now.

**Pull request merged**